Week1/homework/app.js

```diff
@@ -1,11 +1,149 @@
 'use strict';
 
-{
-  const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets'
-  ];
-
-  // Replace with your own code
-  console.log(bookTitles);
+//  1.1
+const bookTitles = [
+  'harry_potter_chamber_secrets',
+  'moll_flanders',
+  'joseph_andrews',
+  'jude_the_obscure',
+  'pride_and_prejudice',
+  'to_sir_with_love',
+  'to_miss_with_love',
+  'where_the_wild_things_are',
+  'me_talk_pretty_one_day',
+  'another_bullshit_night_in_suck_city'
+];
+
+//  1.2
+/*  function createBookList() {
+    const divDom = document.createElement('div');
+    const ulDom = document.createElement('ul');
+    document.body.appendChild(divDom);
+    divDom.appendChild(ulDom);
+    for (let i = 0; i < bookTitles.length; i++) {
+    const liDom = document.createElement('li');
+    ulDom.appendChild(liDom);
+    liDom.setAttribute('id', bookTitles[i]);
+    liDom.textContent = bookTitles[i];
+    }
+  }
+createBookList(); */
+
+//  1.4
+const bookListObj = {
+  harry_potter_chamber_secrets: {
+    title: 'Harry Potter Chamber Secret',
+    language: 'English',
+    author: 'Steve Kloves'
+  },
+
+  moll_flanders: {
+    title: 'Moll Flanders',
+    language: 'English',
+    author: 'Daniel Defoe'
+  },
+  joseph_andrews: {
+    title: 'Joseph Andrews',
+    language: 'English',
+    author: 'Henry Fielding'
+  },
+  jude_the_obscure: {
+    title: 'Jude the Obscure',
+    language: 'English',
+    author: 'Thomas Hardy'
+  },
+  pride_and_prejudice: {
+    title: 'Pride and prejudice',
+    language: 'English',
+    author: 'Jane Austen'
+  },
+  to_sir_with_love: {
+    title: 'To Sir With Love',
+    language: 'English',
+    author: 'E. R. Braithwaite'
+  },
+  to_miss_with_love: {
+    title: 'To Miss With Love',
+    language: 'English',
+    author: 'Katharine Birbalsingh'
+  },
+  where_the_wild_things_are: {
+    title: 'Where the wild Things are',
+    language: 'English',
+    author: 'Maurice Sendak'
+  },
+  me_talk_pretty_one_day: {
+    title: 'Me Talk Pretty One Day',
+    language: 'English',
+    author: 'David Sedaris'
+  },
+  another_bullshit_night_in_suck_city: {
+    title: 'Another Bullshit Night in Suck City',
+    language: 'English',
+    author: 'Nick Flynn'
+  }
+};
+
+//  1.5------1.8
+function getBooksInfo() {
+  const ulDom = document.createElement('ul');
+  document.body.appendChild(ulDom);
+  ulDom.setAttribute('class', 'books');
+  for (let i = 0; i < bookTitles.length; i++) {
+    // const divDom1 = document.createElement('div');
```

Comment:
1. Remove commented-out code when submitting a pull request. The reviewer is not interested in 'dead code'.
2. You can make your code a bit easier to read if you introduce two 'convenience' variables at the top of your loop body:

```js
const bookTitle = bookTitles[i];
const bookDetails = bookListObj[bookTitle];
```

Or, better still, use a `for...of` loop:

```js
for (const bookTitle of bookTitles) {
  const bookDetails = bookListObj[bookTitle];
  // etc
}
```


Now, whenever you use `bookTitles[i]` in your loop body, replace it by `bookTitle`. And where you use `bookListObj[bookTitle]` or even `bookListObj[bookTitles[i]], ` just use `bookDetails`. Less to type, smaller chance to make typing errors.

Reply:
i actully still don't feel comfortable using for ...of loops. but i will give it a try for sure